scaleway/helpers_lb.go

```diff
@@ -43,6 +43,76 @@ func flattenLbBackendMarkdownAction(action lb.OnMarkedDownAction) interface{} {
 	return action.String()
 }
 
+func flattenLbAcl(acl *lb.ACL) map[string]interface{} {
+	res := map[string]interface{}{
+		"name":   acl.Name,
+		"match":  acl.Match,
+		"action": flattenLbAclAction(acl.Action),
+	}
+	return res
+}
+
+// expandLbAcl transforms a state acl to an api one.
+func expandLbAcl(i interface{}) *lb.ACL {
+	rawRule := i.(map[string]interface{})
+	acl := &lb.ACL{
+		Name:   expandOrGenerateString(rawRule["name"], "lb-acl"),
+		Match:  expandLbAclMatch(rawRule["match"]),
+		Action: expandLbAclAction(rawRule["action"]),
+	}
+
+	//remove http filter values if we do not pass any http filter
+	if acl.Match.HTTPFilter == "" || acl.Match.HTTPFilter == lb.ACLHTTPFilterACLHTTPFilterNone {
+		acl.Match.HTTPFilter = lb.ACLHTTPFilterACLHTTPFilterNone
+		acl.Match.HTTPFilterValue = []*string{}
+	}
```

Comment:
if we did pass an HTTPFilter we should make sure it is added in the `acl` variable

```
} else {
    acl.Match.HTTPFilter = lb.ACLHTTPFilterACLHTTPFilter(rawRule["http_filter"].(string))
    acl.Match.HTTPFilterValue = rawRule["http_filter_value"].(string)
}
```

Reply:
It's already taken care of in `expandAclMatch` above. 
```
return &lb.ACLMatch{
		IPSubnet:        ipSubnet,
		HTTPFilter:      lb.ACLHTTPFilter(rawMap["http_filter"].(string)),
		HTTPFilterValue: expandStringsPtr(rawMap["http_filter_value"].([]interface{})),
		Invert:          rawMap["invert"].(bool),
	}
```
The intention for the comparison was to set the httpfilter value to none if nothing has been passed, and to reset filtervalues (if we didn't pass the filter or set it to none but passed filtervalues, which won't be used)